core/routing/routing.go

```diff
@@ -18,24 +18,35 @@ var ErrNotFound = errors.New("routing: not found")
 // type/operation.
 var ErrNotSupported = errors.New("routing: operation or key not supported")
 
-// ContentRouting is a value provider layer of indirection. It is used to find
-// information about who has what content.
-//
-// Content is identified by CID (content identifier), which encodes a hash
-// of the identified content in a future-proof manner.
-type ContentRouting interface {
+// A ContentProvider is able to announce where to find content on the Routing
+// system.
+type ContentProvider interface {
 	// Provide adds the given cid to the content routing system. If 'true' is
 	// passed, it also announces it, otherwise it is just kept in the local
 	// accounting of which objects are being provided.
 	Provide(context.Context, cid.Cid, bool) error
+}
 
+// A ContentProviderFinder is able to retrieve providers for a given CID using
+// the Routing system.
+type ContentProviderFinder interface {
```

Comment:
```suggestion
type ContentDiscovery interface {
```

Reply:
Then should also rename ContentProvider to ContentProviding to have consistent language? Also consisting with PeerRouting

Reply:
done